### ISSUE TITLE
Feature Request: Mean layer implementation

### ISSUE THREAD

**Issue by GPhilo** [opencv/opencv#11038]
- OpenCV => 3.4.0
- Operating System / Platform => Windows 10
- Compiler => MSVC

##### Detailed description

There has been several issues opened on this topic, but -from what I can find- all try to work around the error by removing batch normalization.
Sometimes, however, batch normalization is not the source of the error. As an example, trying to run in opencv the [NASnet mobile](https://github.com/tensorflow/models/tree/master/research/slim/nets/nasnet) model, after solving the problems with `StridedSlice` and `FusedBatchnorm` one eventually gets stuck on the `Mean` layer used before the final fully connected layer, which is used to average the activations on a "per-plane" basis (i.e., if it were applied on a color image, you'd get three values: average R, average B, average G).

From what I could research, there is no workaround for this use of the Mean layer. Considering that NASnet mobile has almost 4% better top-1 performance than mobilenet on Imagenet, it might be relevant to have a working implementation of the Mean layer.

Below I provide minimal python code that generates `graph.pb(txt)?` files with an example of the use-case. I also attach pre-made files generated with tensorflow 1.6 (specific TF version should not be relevant).

##### Steps to reproduce

```.py
import tensorflow as tf
import numpy as np

tf.reset_default_graph()
values = np.asarray([ [ [i]*10 ]*20 for i in range(1,5) ])
c = tf.constant(values)
mean = tf.reduce_mean(c, [1,2])

tf.train.write_graph(tf.get_default_graph().as_graph_def(), '', 'graph.pb', as_text=False)
tf.train.write_graph(tf.get_default_graph().as_graph_def(), '', 'graph.pbtxt', as_text=True)
```

[graph.zip](https://github.com/opencv/opencv/files/1792824/graph.zip)

**Comment by dkurt** [comment]
@GPhilo, please try a PR https://github.com/opencv/opencv/pull/10940.

**Comment by GPhilo** [comment]
Just what I needed, it works like a charm.
